src/duration.js

```diff
@@ -197,8 +198,26 @@ export class Duration {
    * @return {Duration}
    */
   static fromObject(obj) {
+    if (obj == null || typeof obj !== 'object') {
+      return Duration.invalid(INVALID_INPUT);
```

Comment:
This should throw. This is much more likely a programmer error than a data problem. It's similar to how we throw when the user specifies invalid units.